tests/tools.py

```diff
@@ -111,6 +114,47 @@ def verify_all_nodes(self, test_case):
 
     return tested_nodes
 
+  def check_get_text_unmarked(self, node, test_case, text):
+    """
+    Check that `text` (returned from get_text()) usually returns the same text
+    as get_text_unmarked.
+    """
+
+    if (
+        test_case.is_astroid_test
+        or sys.version_info < (3, 8)
+        or 'pypy' in sys.version.lower()
+    ):
+      # These cases are not supported by get_text_unmarked
+      return
+
+    text_unmarked = self.atok.get_text_unmarked(node)
+    if isinstance(node, ast.alias):
+      self._check_alias_unmarked(node, test_case, text_unmarked)
+    elif not isinstance(node, self.bad_unmarked_types):
+      if hasattr(node, 'lineno'):
```

Comment:
How often is 'lineno' absent? (I guess I am worried if it might fail to compare a large set of cases, giving false confidence that `get_text_unmarked` works well.)

Reply:
This is a good point. It's generally for 'unusual' nodes, but apparently that line is blurrier than I thought.